### DISCUSSION TITLE
How to fake Apple Pay payment method in frontend?

### DISCUSSION THREAD

**Discussion by SimonasB88** [cypress-io/cypress#14850] [Component Testing]
Hey, has anyone had experience to somehow mock or stub Apple Pay payment method in frontend?

I have found some community attempts: https://github.com/alphagov/pay-frontend/blob/master/test/cypress/integration/web-payments/apple-pay.cy.test.js

Found stub implementation:
https://github.com/indiegogo/apple-pay-js-stubs

Maybe someone had some experience in stubbing or mocking Apple Pay in frontend, on Electron, Firefox, Chrome, Edge?

Thanks. Very appreciated.

**Comment by 403-html** [comment]
Did you found answer?

**Reply by SimonasB88** [reply]
Nop, still not :)

**Reply by SimonasB88** [reply]
maybe anyone had some experience about mocking Apple Pay? :)

**Reply by lmiller1990** [reply]
Initial link isn't going anywhere: https://github.com/alphagov/pay-frontend/blob/master/test/cypress/integration/web-payments/apple-pay.spec.js

**Reply by SimonasB88** [reply]
@lmiller1990 yea, somehow the URL path has moved to https://github.com/alphagov/pay-frontend/blob/master/test/cypress/integration/web-payments/apple-pay.cy.test.js

**Reply by lmiller1990** [reply]
Thanks for updating this. I didn't get a chance to look at it - it looks like configuring the dev-env is quite involved. I will try to get to this later this week, but I cannot commit to anything right now.

Let me know if you have any progress on your end, I'm definitely interested in solving this problem.

**Reply by EvanSanderson** [reply]
For anyone following along on this thread, the Apple Pay Cypress test example above has moved locations: https://github.com/alphagov/pay-frontend/blob/master/test/cypress/integration/web-payments/apple-pay.test.cy.js

**Comment by maroof-tw** [comment]
Hi There. Did anyone have any success with this? We're implementing apple pay on our website and are looking for ways to implement end-to-end tests for apple pay
@SimonasB88 @lmiller1990

**Reply by 403-html** [reply]
nope, from my side (we need it still)
